src/MenuItemStorage.cpp

```diff
@@ -113,27 +115,40 @@ void MenuItemStorage::render() {
 	Rect disabled_src;
 	disabled_src.x = disabled_src.y = 0;
 	disabled_src.w = disabled_src.h = eset->resolutions.icon_size;
+	bool render;
 
 	for (int i=0; i<slot_number; i++) {
-		if (storage[i].item > 0) {
-			slots[i]->setIcon(items->items[storage[i].item].icon, items->getItemIconOverlay(storage[i].item));
-			slots[i]->setAmount(storage[i].quantity, items->items[storage[i].item].max_quantity);
-		}
-		else {
-			slots[i]->setIcon(WidgetSlot::NO_ICON, WidgetSlot::NO_OVERLAY);
+		render = false;
+
+		if (nb_cols > 0) {
+			render = true;
 		}
-		slots[i]->render();
-		if (!slots[i]->enabled) {
-			if (overlay_disabled) {
-				overlay_disabled->setClipFromRect(disabled_src);
-				overlay_disabled->setDestFromRect(slots[i]->pos);
-				render_device->render(overlay_disabled);
+		else if (nb_cols == 0) {
+			if (menu->inv->isActive(i)) {
```

Comment:
Not a fan of having this sort of circular dependency if it can be avoided. Perhaps a flag on each slot can be used to determine visibility (WidgetSlot has an `enabled` field).

Reply:
As far as i can understand the WidgetSlot `enabled` filed refers to how the slot should be rendered. I noticed that if you have a two handed weapon equipped it will disable the other weapon slot and that disabled slot will still be rendered but in a different way (with a shade over it). If we are going to use this filed for the visibility of the slot then this behavior will change.

What do you say if we add a `bool visible' field to the Widget class to indicate if it should be rendered or not, like the MenuInventory class does here: https://github.com/flareteam/flare-engine/blob/2aeea0c1fe817b532908879315db12c3f42f6095/src/MenuInventory.cpp#L245-L247

Reply:
Ah, you're right. I think a "visible" bool is fine. But I think it only needs to be a member of WidgetSlot instead of Widget.